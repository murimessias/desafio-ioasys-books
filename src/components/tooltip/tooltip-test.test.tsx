import { screen } from '@testing-library/react'
import { renderWithTheme } from 'resources/utils/tests/helpers'

import { Tooltip } from '.'

describe('<Tooltip />', () => {
  it('should render correctly', () => {
    renderWithTheme(<Tooltip error='Error' />)

    expect(screen.getByText('Error')).toHaveStyle({ opacity: 0 })
  })

  it('should render correctly when active', () => {
    const { container } = renderWithTheme(<Tooltip error='Error' active />)

    expect(screen.getByText('Error')).toHaveStyle({ opacity: 1 })

    expect(container.firstChild).toMatchSnapshot()
  })
})
