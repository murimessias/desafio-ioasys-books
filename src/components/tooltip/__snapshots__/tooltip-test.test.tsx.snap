// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tooltip /> should render correctly when active 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #FFFFFF30;
  border-radius: 0.4rem;
  color: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-weight: 500;
  font-size: 1.6rem;
  line-height: 1;
  opacity: 0;
  position: relative;
  padding: 1.6rem 2.4rem;
  -webkit-transition: all 250ms ease;
  transition: all 250ms ease;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  opacity: 1;
}

.c0::before {
  bottom: 100%;
  border-bottom: 0.8rem solid #FFFFFF30;
  border-left: 0.8rem solid transparent;
  border-right: 0.8rem solid transparent;
  content: '';
  height: 0;
  pointer-events: none;
  position: absolute;
  width: 0;
}

<div
  class="c0"
>
  Error
</div>
`
